libnavigation-base/src/test/java/com/mapbox/navigation/base/options/DeviceProfileTest.kt

```diff
@@ -0,0 +1,21 @@
+package com.mapbox.navigation.base.options
+
+import com.mapbox.navigation.testing.BuilderTest
+import io.mockk.mockk
+import kotlin.reflect.KClass
+import org.junit.Test
+
+class DeviceProfileTest : BuilderTest<DeviceProfile, DeviceProfile.Builder>() {
+    override fun getImplementationClass(): KClass<DeviceProfile> = DeviceProfile::class
+
+    override fun getFilledUpBuilder(): DeviceProfile.Builder {
+        return DeviceProfile.Builder()
+            .customConfig("123")
+            .deviceType(mockk(relaxed = true))
```

Comment:
nit: think we should send the non-default DeviceType here, it's a value type and doesn't have a builder covered by tests. enums probably classify as strings or ints

Reply:
Ah it's an enum. We should be fine to compare equality though.